src/openrct2-ui/windows/Ride.cpp

```diff
@@ -2528,35 +2547,42 @@ static rct_string_id window_ride_get_status_station(rct_window* w, Formatter& ft
         return STR_NONE;
 
     int32_t count = w->ride.view - ride->num_vehicles - 1;
-    StationIndex stationIndex = STATION_INDEX_NULL;
     rct_string_id stringId = STR_EMPTY;
+    int32_t stationIndex = 0;
+    auto lastStation = ride->stations[0];
 
-    do
+    for (const auto& station : ride->stations)
```

Comment:
The whole reason for making a helper function is to use it. Replace this bit with the new helper function.